erpnext/public/js/controllers/transaction.js

```diff
@@ -908,14 +913,50 @@ erpnext.TransactionController = erpnext.taxes_and_totals.extend({
 				doc: this.frm.doc,
 				method: "apply_shipping_rule",
 				callback: function(r) {
-					if(!r.exc) {
+					if(!r.exc && localStorage) {
+						let last_shipping_rule = localStorage.getItem("last_shipping_rule");
+						if(last_shipping_rule && me.frm.doc.shipping_rule != last_shipping_rule)	{
+							me.remove_duplicate_taxes(last_shipping_rule,"shipping_rule");
+						}
+						me.set_shipping_last_value();
 						me.calculate_taxes_and_totals();
 					}
 				}
 			}).fail(() => this.frm.set_value('shipping_rule', ''));
-		}
-		else {
-			me.calculate_taxes_and_totals();
+		}		
+	},
+
+	remove_duplicate_taxes: function(last_shipping_rule,field_type) {
+			return this.frm.call({
```

Comment:
@hrwX is it a good practice to return a function from here? I've never seen a function return another function in the js files that we have, but I may be wrong